.github/workflows/sbom-matrix.yaml

```
@@ -0,0 +1,409 @@
+  # them to a Dependency Track instance for vulnerability scanning and analysis.
+  # The upload only happens for specific branches (ionos-stable and the feature branch).
+  #
+  # The job:
+  # 1. Downloads all SBOM artifacts from the generate-sbom job
+  # 2. Iterates through each component in the matrix
+  # 3. Determines which SBOM file to upload (combined, composer-only, or npm-only)
+  # 4. Uploads each SBOM to the corresponding Dependency Track project
+  # 5. Uses custom CA certificate for secure communication with Dependency Track
+  upload-sboms:
+    needs: [ get-version, setup-matrix, generate-sbom ]
+    runs-on: self-hosted
+
+    steps:
+      - name: Download all SBOM artifacts
+        uses: actions/download-artifact@v4
+        with:
+          pattern: sbom-*
+          merge-multiple: true
+
+      - name: List downloaded files
+        run: |
+          ls -la *.json || echo "No BOM JSON files found"
+
+      - name: Upload SBOMs to Dependency Track
+        if: github.ref == 'refs/heads/ionos-dev'
+        env:
+          DT_BASE_URL: ${{ vars.DEPENDENCY_TRACK_BASE_URL }}
+          DT_API_KEY: ${{ secrets.DEPENDENCY_TRACK_API_KEY }}
+          DT_ROOT_OBJECT_ID: ${{ vars.DT_ROOT_OBJECT_ID }}
+          IONOS_CA_CERT: ${{ secrets.IONOS_CA }}
+          MATRIX_CONTEXT: ${{ needs.setup-matrix.outputs.matrix }}
+          PROJECT_VERSION: ${{ needs.get-version.outputs.project_version }}
+          VARS_CONTEXT: ${{ toJSon(vars) }}
+        run: |
+          # Create temporary CA cert file
+          cert_file=$(mktemp)
+          echo "$IONOS_CA_CERT" > "${cert_file}"
+          
+          echo "Beginning SBOM upload process..."
+          echo "PROJECT_VERSION: $PROJECT_VERSION"
+          
+          # Function to upload SBOM to Dependency Track via REST API
+          upload_bom() {
+            local bom_file="$1"
+            local project_name="$2"
+            local artifact_type="$3"
+            local qualified_artifact_name="$2 $3"
+            
+            if [[ ! -f "$bom_file" ]]; then
+              echo "Warning: $bom_file not found, skipping..."
+              return 0
+            fi
+            
+            echo "Uploading SBOM ($bom_file) to parent project ${DT_ROOT_OBJECT_ID} ($project_name and artifact type $artifact_type)..."
+          
+            response=$(curl \
+              --cacert "${cert_file}" \
+              --fail-with-body \
+              --silent \
+              --show-error \
+              --write-out "HTTPSTATUS:%{http_code}" \
+              -X POST "${DT_BASE_URL}/api/v1/bom" \
+              -H "Content-Type: multipart/form-data" \
+              -H "X-API-Key: ${DT_API_KEY}" \
+              -F "isLatest=true" \
+              -F "autoCreate=true" \
+              -F "parentUUID=${DT_ROOT_OBJECT_ID}" \
+              -F "projectName=${qualified_artifact_name}" \
+              -F "projectVersion=${PROJECT_VERSION}" \
```

```suggestion
              -F "parentUUID=${DT_ROOT_OBJECT_ID}" \
              -F "projectName=${qualified_artifact_name}" \
              -F "projectTags=hidrive_next,nextcloud" \
              -F "projectVersion=${PROJECT_VERSION}" \
```

lets keep tags in order not to rely on only on project title